fd/diff.go

```diff
@@ -98,7 +108,8 @@ func Derivative(f func(float64) float64, x float64, settings *Settings) float64
 // otherwise a new slice will be allocated and returned. Finite difference
 // kernel and other options are specified by settings. If settings is nil,
 // default settings will be used.
-// Gradient panics if the length of dst and x is not equal.
+// Gradient panics if the length of dst and x is not equal, or if the
```

Comment:
I like that. ```func Hessian(*mat64.SymDense, stuff) *mat64.SymDense```` and ```func Jacobian(*mat64.Dense, stuff) *mat64.Dense``` would both be very useful. If nothing else, it would be a convenient place to look up if the Jacobian is df_i / dx_j or df_j / dx_i :)